src/net_processing.cpp

```diff
@@ -468,6 +468,44 @@ static CNodeState *State(NodeId pnode) EXCLUSIVE_LOCKS_REQUIRED(cs_main) {
     return &it->second;
 }
 
+/**
+ * Data stucture for an individual peer. This struct is not protected by
+ * cs_main since it does not contain validation-critical data.
+ *
+ * Memory is owned by shared pointers and this object is destructed when
+ * the refcount drops to zero (in FinalizeNode()).
```

Comment:
nit in commit d60a16bb3f58fcb9422c277256b3c3a31cab3b2f
```suggestion
 * the refcount drops to zero.
```

FinalizeNode only decreases the refcount by one. It can still be larger than one (as long as someone is holding a PeerRef, e.g. `getpeerinfo` or similar)

Reply:
Fixed